*******************************************************************************
Date         : 2019-07-16
Author       : Yi Wang
Harman International Industries, Incorporated

Description  : This patch is for fixing mtp sync can't complete issue
ChangeList   : 5415782
*******************************************************************************

diff --git a/src/simple-mtpfs-fuse.cpp b/src/simple-mtpfs-fuse.cpp
index c51e605..7c84d80 100755
--- a/src/simple-mtpfs-fuse.cpp
+++ b/src/simple-mtpfs-fuse.cpp
@@ -158,7 +158,7 @@ int wrap_opendir(const char *path, struct fuse_file_info *file_info)
 int wrap_readdir(const char *path, void *buf, fuse_fill_dir_t filler,
     off_t offset, struct fuse_file_info *file_info)
 {
-    LogDebug("enter wrap_readdir, path:%s, offset:%d", path, offset);
+    LogDebug("enter wrap_readdir, path:%s, offset:%lld", path, offset);
     return SMTPFileSystem::instance()->readdir(path, buf, filler,
         offset, file_info);
 }
diff --git a/src/simple-mtpfs-mtp-device.cpp b/src/simple-mtpfs-mtp-device.cpp
index 0b35f81..24581a1 100755
--- a/src/simple-mtpfs-mtp-device.cpp
+++ b/src/simple-mtpfs-mtp-device.cpp
@@ -295,6 +295,7 @@ const TypeDir *MTPDevice::dirFetchContentFromCache(std::string path)
         int count = 0;
         for (LIBMTP_devicestorage_t *s = m_device->storage; s; s = s->next) {
             m_root_dir.addDir(TypeDir(s_root_node, 0, s->id, std::string(s->StorageDescription)));
+            LogInfo("path:%s addDir %s, storageid:%d", path.c_str(), s->StorageDescription, s->id);
             LogDebug("path:%s setFetched", path.c_str());
             m_root_dir.setFetched();
             count++;
@@ -345,7 +346,7 @@ void* dirFetchCountThreadRUn(void *arg)
       return NULL;
    }
 
-   LogInfo("enter  dirFetchCountThreadRUn dir name:%s", dir->name().c_str());
+   LogInfo("enter  dirFetchCountThreadRUn dir name:%s storageid: %d, id:%d", dir->name().c_str(), dir->storageid(), dir->id());
 
    if(g_device == NULL)
    {
diff --git a/src/simple-mtpfs-type-basic.h b/src/simple-mtpfs-type-basic.h
index b6b1b31..0bb33a3 100755
--- a/src/simple-mtpfs-type-basic.h
+++ b/src/simple-mtpfs-type-basic.h
@@ -56,12 +56,12 @@ public:
         m_parent_id(copy.m_parent_id),
         m_storage_id(copy.m_storage_id),
         m_name(copy.m_name),
-        fetched_offset(0),
-        item_n(0),
-        fetched_handle_done(false)
+        fetched_offset(copy.fetched_offset),
+        item_n(copy.item_n),
+        fetched_handle_done(copy.fetched_handle_done)
     {
-       handles.n = 0;
-       handles.Handler = NULL;
+       handles.n = copy.handles.n;
+       handles.Handler = copy.handles.Handler;
     }
 
     uint32_t id() const { return m_id; }
